### DISCUSSION TITLE
Add the "robots: noindex" meta to all pages inside a directory

### DISCUSSION THREAD

**Discussion by AnastasiaTWW** [squidfunk/mkdocs-material#6634] [Q&A] [answered]
I need to add the "robots: noindex" meta to all pages inside a directory docs/2.18. I've found this way https://squidfunk.github.io/mkdocs-material/reference/#using-metadata-in-templates and also tried to use the `meta` plugin in the following format:

meta:
  - name: robots
    content: noindex, nofollow

However, no success. Probably someone faced with the same or just knows how to do that?

**Comment by kamilkrzyskow** [comment]
Hello @AnastasiaTWW,
please provide a minimal example of what you're doing: https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/

The only thing I can suggest now is to check for typos in your `main.html` override of the `extrahead` block.
https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme
https://github.com/squidfunk/mkdocs-material/blob/ab5ccbe6b8c49dbf58a25eb066a98b243a382967/src/templates/base.html#L171-L175

Also AFAIK the `meta` plugin doesn't add HTML `<meta>` tags to the source it just adds it the the front-matter of the page, so that way you can use it inside the `main.html` override.
Also I'm not sure how to understand "no success" here. Are you checking the generated HTML source for the new meta tags or do you check if the robots tag is detected with some tool 🤔

**Reply by squidfunk** [reply, marked as answer]
The canonical way to do this is to use [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme), add the tags you want to add to `extrahead`, and then use that template with [`template`](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-template). You can then set `template` with the meta plugin for an entire subsection of your docs.

**Reply by AnastasiaTWW** [reply]
Thank a lot! It worked
